app/models/user.rb

```diff
@@ -103,8 +103,35 @@ def decline_invite
     update(invited_by_id: nil)
   end
 
-  private
-    def email_without_domain
-      email.split('@').first
+  def process_unit(unit_params, user_is_guest)
+    if user_is_guest # if user is guest
+      if unit = self.unit # and has a unit
```

Comment:
Yes, I understand.  Here is what it looks like using your suggestion. I had to add more lines and it ended up being the exact same number of lines altogether. What am I missing? I admit it looks repetitive, but I haven't been able to refactor this while keeping the desired behavior.

```
def process_unit(unit_params, user_is_guest)
    if user_is_guest && unit = self.unit # if user is guest and has a unit
      bill = Bill.find_by(unit: unit)
      bill.delete if bill != nil # delete guest's previously entered bills
      self.update(unit: nil) # delete guest's previously entered unit
      unit.destroy
      return self.assign_new_unit(unit_params)
    elsif user_is_guest #if user is guest and does not have a unit
      return self.assign_new_unit(unit_params)
    elsif unit = self.unit # user is not guest and has unit
      return unit
    else # user is not guest and no unit
      return self.assign_new_unit(unit_params)
    end
end
```